src/test/java/by/itechart/api/controller/UserControllerIT.java

```diff
@@ -0,0 +1,120 @@
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@WithMockUser(username = "admin@admin.com", roles = "ADMIN", password = "$2a$12$mbrKKshlIeJQNszD4KJSjeXCAZGbIMrcDG8qkiPkwTXy2G4dNXzrW")
+@SqlGroup({
+        @Sql(value = "/db.script/add_users_before.sql", executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD),
+        @Sql(value = "/db.script/delete_all_data_after.sql", executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+})
+class UserControllerIT {
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private WebApplicationContext context;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private final Long USER_ID = 2L;
+
+    @Test
+    void getAllUsers() throws Exception {
+        this.mockMvc.perform(get("/users/all")).andDo(print()).andExpect(status().isOk())
+                .andExpect(jsonPath("$.*.id", hasSize(2)));
+    }
+
+    @Test
+    void getCurrentUser() throws Exception {
+        this.mockMvc.perform(get("/users/current")).andDo(print()).andExpect(status().is(302))
+                .andExpect(jsonPath("$.email", is("admin@admin.com")));
+    }
+
+    @Test
+    void updateCurrentUser() throws Exception {
+        var updateUserDTO = getUpdateUserDTO();
+        this.mockMvc.perform(patch("/users/current")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(updateUserDTO))
+                .with(csrf()))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.email", is("newemail@email.com")))
+                .andExpect(jsonPath("$.phone", is("3213213213")));
+    }
+
+    @Test
+    void updateUser() throws Exception {
+        var updateUserDTO = getUpdateUserDTO();
+        this.mockMvc.perform(patch("/users/{userID}", USER_ID)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(updateUserDTO))
+                .with(csrf()))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.email", is("newemail@email.com")))
+                .andExpect(jsonPath("$.phone", is("3213213213")));
+    }
+
+    @Test
+    void deleteUser() throws Exception {
+        this.mockMvc.perform(delete("/users/{userID}", USER_ID).with(csrf())).andDo(print())
```

Comment:
You can also add one test to check that user without `ADMIN` role will be declined to perform this action